### PR TITLE
docs: update release process

### PR DESCRIPTION
**Pull request by dalelane**
Updates to the document used by the release coordinator
to help smooth the process for the next release.

Contributes to: asyncapi/spec#675

Signed-off-by: Dale Lane <dale.lane@uk.ibm.com>

### CODE REVIEW

File: RELEASE_PROCESS.md
Comment (dalelane) [inline RELEASE_PROCESS.md:267]: @derberg Can you help fill me in on how this works, please?

File: .github/ISSUE_TEMPLATE/release.md
Comment (dalelane) [inline .github/ISSUE_TEMPLATE/release.md:3]:
```suggestion
about: Track the progress of the next release of the AsyncAPI spec
```

**Comment by dalelane** [comment]
I think there are more things that we should include for website updates - inspired by https://github.com/asyncapi/website/pull/614 but from a quick search I can see other places that would justify attention, such as the examples in https://github.com/asyncapi/website/blob/master/pages/docs/getting-started/hello-world.md

File: RELEASE_PROCESS.md
Comment (derberg) [inline RELEASE_PROCESS.md:254]: in `#specification` channel

File: RELEASE_PROCESS.md
Comment (derberg) [inline RELEASE_PROCESS.md:290]: it is more about updating existing release created by bot rather than creating a release

File: RELEASE_PROCESS.md
Comment (derberg) [inline RELEASE_PROCESS.md:295]: we could do this also when release cycle is started, to let folks know upfront, and what issue to follow to see progress

Comment (derberg) [reply RELEASE_PROCESS.md:295]: you can do it by basically mentioning codeowners in the release progress issue

File: RELEASE_PROCESS.md
Comment (derberg) [inline RELEASE_PROCESS.md:306]: please advice that release coordinator should be proud of the work done, and announce release from private accounts on Twitter and LinkedIn and then these will be used as a base for official announcement to assure release coordinator participation

File: RELEASE_PROCESS.md
Comment (derberg) [reply RELEASE_PROCESS.md:267]: Once PR with fix or feat prefix is merged into release branch, automation and bot produces a prerelease, release candidate that includes in name upcoming release number, date of planned release and also the number of the candidate. Example https://github.com/asyncapi/spec/releases/tag/v2.3.0-2022-01-release.3

Important to know is that if `@asyncapi/specs` candidate is released, then PP in Parser-js for the same feature should be updated to use the release candidate, so the released release candidate for parser actually contains related changes in JSON Schema

**Review by derberg — COMMENTED** [review]
@dalelane sorry for being so late with review, added some comments

File: RELEASE_PROCESS.md
Comment (smoya) [inline RELEASE_PROCESS.md:89]: I suggest we follow semantic release `next` and `next-major` branch names.

I suggested it in the past in Slack as well: https://asyncapi.slack.com/archives/C0230UAM6R3/p1644418043568169

Comment (smoya) [reply RELEASE_PROCESS.md:89]: I ended up creating an issue for my suggestion: https://github.com/asyncapi/spec/issues/734

File: .github/ISSUE_TEMPLATE/release.md
Comment (smoya) [inline .github/ISSUE_TEMPLATE/release.md:26]:
```suggestion
- [ ] Create release branches
```

You will need to add an extra space in all checkboxes to be rendered properly.

File: RELEASE_PROCESS.md
Comment (dalelane) [reply RELEASE_PROCESS.md:108]: it's just a link to where the term is defined in the page

File: RELEASE_PROCESS.md
Comment (smoya) [inline RELEASE_PROCESS.md:113]: Nitpick: Could we prefix the subtitles with a step number, such as `4.1`, `4.2` etc? I'm following this guide right now and I missed that info which would make it easier for me to visualize the steps to follow.

Comment (dalelane) [reply RELEASE_PROCESS.md:113]: good idea 👍

**Comment by derberg** [comment]
@dalelane I see you wanted to merge. It could not work as there were no approval. When you are done with updating PR basing on provided feedback remember about hitting the "arrows" icon that rerequests review and triggers notification to reviewers.

Screenshot 2022-03-17 at 12 22 42

anyway, great job and now I approve 🚀 
you can try merge again 😄

**Review by derberg — APPROVED** [review]

**Comment by derberg** [comment]
@fmvilas not sure if you had a chance to see it so pinging you so you can see great job from Dale on improving the release process

**Comment by smoya** [comment]
This is awesome it finally got merged 👏 Awesome stuff @dalelane ! thanks

**Comment by derberg** [comment]
now with each release goal should be to not add more steps, but remove as many manual steps as possible one by one 😅

**Comment by fmvilas** [comment]
Great stuff @dalelane 👏

**Pull request merged**
